src/test/java/de/tum/in/www1/artemis/ExamIntegrationTest.java

```diff
@@ -1801,24 +1814,123 @@ public void testGetStatsForExamAssessmentDashboard() throws Exception {
+
+        for (var exercise : exercisesInExam) {
+            for (var participation : exercise.getStudentParticipations()) {
+                Submission submission;
+                assertThat(participation.getSubmissions()).hasSize(1);
+                submission = participation.getSubmissions().iterator().next();
+                // Create results
+                var result = new Result().score(50D).rated(true).resultString("Good");
+                if (exercise instanceof QuizExercise) {
+                    result.completionDate(ZonedDateTime.now().minusMinutes(3));
+                }
+                result.setAssessmentType(AssessmentType.SEMI_AUTOMATIC);
+                result.setParticipation(participation);
+                result.setAssessor(examInstructor);
+                result = resultRepository.save(result);
+                result.setSubmission(submission);
+                submission.addResult(result);
+                submissionRepository.save(submission);
+            }
+        }
+        // check the stats again
+        database.changeUser("instructor1");
+        var stats = request.get("/api/courses/" + course.getId() + "/exams/" + exam.getId() + "/stats-for-exam-assessment-dashboard", HttpStatus.OK, StatsForDashboardDTO.class);
+        assertThat(stats.getNumberOfAssessmentLocks()).isEqualTo(75L);
+        // 75 = (15 users * 5 exercises); quiz submissions are not counted
+        assertThat(stats.getNumberOfSubmissions().getInTime()).isEqualTo(75L);
+        // the 15 quiz submissions are already assessed - and all are assessed in teh first correctionRound
+        assertThat(stats.getNumberOfAssessmentsOfCorrectionRounds()[0].getInTime()).isEqualTo(90L);
+        assertThat(stats.getNumberOfAssessmentsOfCorrectionRounds()[1].getInTime()).isEqualTo(15L);
+        assertThat(stats.getNumberOfComplaints()).isEqualTo(0L);
+        assertThat(stats.getTotalNumberOfAssessmentLocks()).isEqualTo(75L);
+
+        // test the query needed for assessment information
+        database.changeUser("tutor2");
+        exam.getExerciseGroups().forEach(group -> {
+            var locksRound1 = group.getExercises().stream().map(
+                    exercise -> resultRepository.countNumberOfLockedAssessmentsByOtherTutorsForExamExerciseForCorrectionRounds(exercise, numberOfCorrectionRounds, examTutor2)[0]
+                            .getInTime())
+                    .reduce((x, y) -> x + y).get();
+            if (group.getExercises().stream().filter(exercise -> !(exercise instanceof QuizExercise)).count() != 0)
+                assertThat(locksRound1).isEqualTo(0L);
+
+            var locksRound2 = group.getExercises().stream().map(
+                    exercise -> resultRepository.countNumberOfLockedAssessmentsByOtherTutorsForExamExerciseForCorrectionRounds(exercise, numberOfCorrectionRounds, examTutor2)[1]
+                            .getInTime())
+                    .reduce((x, y) -> x + y).get();
```

Comment:
please define a more readable variable instead of x and y :)

Reply:
Done, seems like that was a leftover from writing the tests, good catch